android/src/main/java/com/reactnativekeyboardcontroller/views/overlay/JSPointerDispatcherCompat.kt

```diff
@@ -0,0 +1,36 @@
+package com.reactnativekeyboardcontroller.views.overlay
+
+import android.view.MotionEvent
+import android.view.ViewGroup
+import com.facebook.react.uimanager.JSPointerDispatcher
+import com.facebook.react.uimanager.events.EventDispatcher
+
+/**
+ * Compat layer for `JSPointerDispatcher` interface for RN < 0.72
+ */
+class JSPointerDispatcherCompat(
+  private val viewGroup: ViewGroup,
+) : JSPointerDispatcher(viewGroup) {
+  fun handleMotionEventCompat(
+    event: MotionEvent?,
+    eventDispatcher: EventDispatcher?,
+    isCapture: Boolean,
+  ) {
+    val method =
+      try {
+        JSPointerDispatcher::class.java.getMethod(
+          "handleMotionEvent",
+          MotionEvent::class.java,
+          EventDispatcher::class.java,
+          Boolean::class.javaPrimitiveType,
+        )
+      } catch (e: NoSuchMethodException) {
+        JSPointerDispatcher::class.java.getMethod(
+          "handleMotionEvent",
+          MotionEvent::class.java,
+          EventDispatcher::class.java,
+        )
+      }
+    method.invoke(this, event, eventDispatcher, isCapture)
```

Comment:
Does this code works as expected (without exceptions) on RN < 0.71?

Reply:
https://github.com/facebook/react-native/pull/33526/files#diff-e874545c1f508ac02d63d67356fe0519f6c0dd5f380afeb900cf1de4fce6835a

If it goes to the location where the file was initially created, it should work well since the arguments will be two from the start. However, it is challenging to directly test this code. I need a scenario to test it.

Reply:
In case `getMethod` fails to retrieve the method, it likely indicates an even older version, in which case it shouldn’t execute. So, I adjusted the position of the `method.invoke` code.

Reply:
Well, if you want to test your code, you can take this screen https://github.com/kirillzyusko/react-native-keyboard-controller/tree/main/example/src/screens/Examples/OverKeyboardView and do following:

- set focus to input
- wait for keyboard to appear
- press "Show" (it will show blue rectangle)
- click on blue rectangle (it should hide blue rectangle)
- click on "Show" again (it will show blue rectangle again)
- click outside of blue rectangle
- it should close it

If this scenario works on RN < 0.71, then it's safe to merge 😊

Reply:
It works well in my environment (version 0.71.14). Should I also check for compatibility with previous versions?


https://github.com/user-attachments/assets/b131dfdb-fc37-4708-9604-fc0a0999e835

Reply:
@gronxb no, I don't think we need to test versions below 0.71. Thank you for testing, your PR and quick responses, I appreciate it! Let's merge this PR - code looks good for me 😊